firebase_admin/functions.py

```diff
@@ -0,0 +1,409 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Firebase Functions module."""
+
+from __future__ import annotations
+from datetime import datetime, timedelta
+from urllib import parse
+import re
+import json
+from base64 import b64encode
+from typing import Any, Optional, Dict
+from dataclasses import dataclass
+from google.auth.compute_engine import Credentials as ComputeEngineCredentials
+
+import requests
+import firebase_admin
+from firebase_admin import App
+from firebase_admin import _http_client
+from firebase_admin import _utils
+from firebase_admin.exceptions import FirebaseError
+
+_FUNCTIONS_ATTRIBUTE = '_functions'
+
+__all__ = [
+    'TaskOptions',
+
+    'task_queue',
+]
+
+
+_CLOUD_TASKS_API_RESOURCE_PATH = \
+    'projects/{project_id}/locations/{location_id}/queues/{resource_id}/tasks'
+_CLOUD_TASKS_API_URL_FORMAT = \
+    'https://cloudtasks.googleapis.com/v2/' + _CLOUD_TASKS_API_RESOURCE_PATH
+_FIREBASE_FUNCTION_URL_FORMAT = \
+    'https://{location_id}-{project_id}.cloudfunctions.net/{resource_id}'
+
+_FUNCTIONS_HEADERS = {
```

Comment:
I think we can move some of these constants out of the global scope and into a mode limited scope. For example, can we defined `_FUNCTIONS_HEADERS` inside the `TaskQueue` class?

Reply:
Actually, ignore this. I think the module level constants are okay!